gateway/src/main/java/org/georchestra/gateway/app/GeorchestraGatewayApplication.java

```diff
@@ -0,0 +1,87 @@
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * geOrchestra.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.georchestra.gateway.app;
+
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.cloud.gateway.route.RouteLocator;
+import org.springframework.context.event.EventListener;
+import org.springframework.core.env.Environment;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClient;
+import org.springframework.security.oauth2.client.annotation.RegisteredOAuth2AuthorizedClient;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.util.unit.DataSize;
+import org.springframework.web.bind.annotation.GetMapping;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Controller
+@Slf4j
+@SpringBootApplication
+public class GeorchestraGatewayApplication {
+
+    private @Autowired RouteLocator routeLocator;
+
+    public static void main(String[] args) {
+        SpringApplication.run(GeorchestraGatewayApplication.class, args);
+    }
+
+    @GetMapping("/testme")
+    public String index(Model model, @RegisteredOAuth2AuthorizedClient OAuth2AuthorizedClient authorizedClient,
+            @AuthenticationPrincipal OAuth2User oauth2User) {
+        String name = oauth2User.getName();
+        String clientName = authorizedClient.getClientRegistration().getClientName();
+        Map<String, Object> attributes = oauth2User.getAttributes();
+
+        model.addAttribute("userName", name);
+        model.addAttribute("clientName", clientName);
+        model.addAttribute("userAttributes", attributes);
+        return "index";
+    }
+
+    @EventListener(ApplicationReadyEvent.class)
+    public void onApplicationReady(ApplicationReadyEvent e) {
+        Environment env = e.getApplicationContext().getEnvironment();
+        String datadir = env.getProperty("georchestra.datadir");
+        String app = env.getProperty("spring.application.name");
+        String instanceId = env.getProperty("info.instance-id");
+        int cpus = Runtime.getRuntime().availableProcessors();
+        String maxMem;
+        {
+            DataSize maxMemBytes = DataSize.ofBytes(Runtime.getRuntime().maxMemory());
+            double value = maxMemBytes.toKilobytes() / 1024d;
+            String unit = "MB";
+            if (maxMemBytes.toGigabytes() > 0) {
+                value = value / 1024d;
+                unit = "GB";
+            }
+            maxMem = String.format("%.2f %s", value, unit);
+        }
```

Comment:
I'm curious but what the extra block provides ? Is it for readability, or garbage-collecting purposes ?